docs/DeveloperGuide.md

```diff
@@ -196,10 +196,131 @@ H  I     J  K      L  M      N  O
+4. This results in the method to call for another nested method - `getGate("and")` in order to obtain the gate of the template type
+specified by the user in their input. The method will then return an `AndGate()` object to this method call.
+5. This then results in the instantiation and return of the `BooleanTemplate(AndGate())` which is the boolean template of an `and` gate back to the `BooleanParser` object.
+6. `TemplateBooleanCommand(BooleanTemplate(AndGate()))` is then called in the `TemplateBooleanCommand` object.
+7. By inheritance of superclass, the `Command` object can be instantiated and by `super(BooleanTemplate(AndGate()))` in the `TemplateBooleanCommand` object, this template
+obtained is assigned to the `this.template` in the `Command` object.
+
+The aforementioned sequence of events can be represented in the following sequence diagram:
+
+![InsertClassDiagram](diagrams/TemplateBooleanCommand.png)
+
+#### `SetBooleanCommand`
+The `setBooleanCommand` is used to set the value of one of two inputs to the logic gates, which can be either the Boolean value of `0 or 1`.
+
+The sequence by which the `CalcBooleanCommand` is instantiated is as follows:
+
+1. It is instantiated through the `execute()` function in the Parser class.
+2. The `setBooleanCommand` object will then call the `setInput(template logicGate, String gateInput, int value)` method which will take in a template, input of a logic gate as `String` data type, 
+and its Boolean value to be set as its three inputs. The function takes in a template as its first input and will instantiate the BooleanTemplate object.
+3. The `BooleanTemplate` object will then convert the input of the logic gate into the index to be stored in the binary tree of logic gates. This will depend on where the 
+logic gate is stored in the binary tree. This can be achieved by passing in the gate in `String` form to the `convertToIndex("C")` function, where in this case `"C"` is the
+input of the logic gate to be converted.
+4. The `BooleanTemplate` object will then call `insertTree` function in the instantiated `BinaryTree` object, where the `insertTree` function will take in the index of the logic gate
+stored in the `BinaryTree` and the Boolean value to be assigned to the gate as input. The binary tree containing the logic gates is stored in this object.
+5. The `BinaryTree` object will then call on the `setInput` function in the instantiated `OrGate` object with `index` of the logic gate and its Boolean value to be
+assigned as inputs.
+6. The `OrGate` object represents the `or` operation logic gate. 
+It then calls `setBoolean(int value)` function on itself to assign the Boolean value to that selected input of the logic gate.
+
+The above sequence of object interactions through the SetBooleanCommand can be represented in the following sequence diagram:
+
+![InsertSequeunceDiagram](diagrams/SetBooleanCommand.png)
```

Comment:
This is still not fixed. Do change the name of the image file to `SetBooleanCommand.png`.

Reply:
Okay but it is already `SetBooleanCommand.png` right?

Reply:
The name of the actual file is still `docs/diagrams/setBooleanCommand.png`. It will not render properly if the case does not match.

Reply:
Ah, sorry for the confusion. I changed it to `SetBooleanCommand.png` already but didn't push it yet.